src/problem_description.cpp

```diff
@@ -42,6 +42,18 @@ bool ProblemDescription::IsLayoutDefault() const
     }
 }
 
+bool ProblemDescription::IsLayoutNHWC() const
+{
+    if(spatial_dims == 2)
+    {
+        return (in_layout == "NHWC") && (out_layout == "NHWC") && (weights_layout == "NHWC");
```

Comment:
IIRC the layouts for the input and weights are usually naturally derived from the input layout. Thus, for the NCHW input, the output and weight will always be KCHW and KCYX.

It looks terrible to me that values of `out_layout` and `weights_layout` are different from actual layouts. Do we have any  plans what to do with this?

Are we ever going to support "unnatural" layout combinations? If not, can we just get rid of `out_layout` and `weights_layout`?

Reply:
I don't know about if reference kernel support "unnatural" layouts. However, MLIR backend solvers does give us the capability to do "unnatural" layouts convolution. Unless there's strong opposition I hope to still keep the `out-layout` and `weights_layout`.

I'd hope to address the letters as more natural one as much as you think but I think that's not possible without addressing this [issue 509](https://github.com/ROCmSoftwarePlatform/MIOpen/issues/509) first.

Reply:
reference kernel can support different layout, but that still need write by hand. There is no optimization, nor template usage in reference kernel, everything is just to keep it as simple as possible, so in this case reference kernel do lack flexibility compared with MLIR.

So I think, maybe keep check `out_layout` 'weights_layout' here? because we indeed only support this kind of scenario, for now.

Reply:
I think it's good to not having assumption on the output layout based on input and weight

Reply:
> Thus, for the NCHW input, the "natural" output and weight will be KCHW and KCYX.

Well, actually tensor layout becomes "KCHW" or "KCYX" only in the context of convolution. But tensors do not depend on the operations in which they are supposed to be used.

Reply:
So,
> It looks terrible to me that values of out_layout and weights_layout are different from actual layouts. Do we have any plans what to do with this?

we can't do anything with this for now.